plugins/aggregators/basicstats/basicstats.go

```diff
@@ -139,13 +143,15 @@ func (b *BasicStats) Add(in telegraf.Metric) {
 				//sum compute
 				tmp.sum += fv
 				//diff compute
-				tmp.diff = fv - tmp.LAST
+				tmp.diff = fv - tmp.PREVIOUS
 				//interval compute
 				tmp.interval = in.Time().Sub(tmp.TIME)
 				//rate compute
 				if !in.Time().Equal(tmp.TIME) {
 					tmp.rate = tmp.diff / tmp.interval.Seconds()
 				}
+				//last compute
+				tmp.last = fv
```

Comment:
Currently this tracks `last` based on metric aggregation order. Should this instead track by latest metric timestamp?

Reply:
Good question. I think the current behavior you have here, which is what was previously for tracking and computation is the correct behavior.